packages/core/src/httpClient/basicHttpClient.ts

```diff
@@ -253,12 +270,12 @@ export function headersWithDefaultField(
   return headers;
 }
 
-export interface HttpRequest extends HttpRequestOptions {
+export type HttpRequest<ExtraOptions = any> = {
   path: string;
   method: HttpMethod;
-}
+} & HttpRequestOptions<ExtraOptions>;
```

Comment:
An interface cannot extend another interface with unknown members - so now `HttpRequest` is a type.

Reply:
Can we remove the `any` part? Then it should be extendable.
`ExtraOptions = any` is scary. We don't know how it looks like (not even if it is an object). We should control what are the extra options someone can pass in here.

Reply:
I agree the comment, but removed that part directly so no longer applies. Instead of making `HttpRequest` generic, extending it in `retryableHttpClient.ts` file also works